pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -85,17 +153,31 @@ stdenv.mkDerivation rec {
   cmakeDir = "../cmake";
 
   cmakeFlags = [
-    "-Donnxruntime_PREFER_SYSTEM_LIB=ON"
+    "-DCMAKE_BUILD_TYPE=RELEASE"
+    "-DFETCHCONTENT_FULLY_DISCONNECTED=ON"
+    "-DFETCHCONTENT_QUIET=OFF"
+    "-DFETCHCONTENT_SOURCE_DIR_ABSEIL_CPP=${abseil-cpp_202206.src}"
+    "-DFETCHCONTENT_SOURCE_DIR_DATE=${howard-hinnant-date}"
+    "-DFETCHCONTENT_SOURCE_DIR_EIGEN=${eigen}"
+    "-DFETCHCONTENT_SOURCE_DIR_FLATBUFFERS=${flatbuffers}"
+    "-DFETCHCONTENT_SOURCE_DIR_GOOGLETEST=${gtest}"
+    "-DFETCHCONTENT_SOURCE_DIR_GSL=${gsl}"
+    "-DFETCHCONTENT_SOURCE_DIR_MICROSOFT_WIL=${wil}"
+    "-DFETCHCONTENT_SOURCE_DIR_MP11=${mp11}"
+    "-DFETCHCONTENT_SOURCE_DIR_ONNX=${python3Packages.onnx.src}"
+    "-DFETCHCONTENT_SOURCE_DIR_PYTORCH_CPUINFO=${pytorch_cpuinfo}"
+    "-DFETCHCONTENT_SOURCE_DIR_SAFEINT=${safeint}"
```

Comment:
I tried to reuse sources from `nixpkgs` where possible but the versions currently available are either too old (`wil` and `gtest` from the top of my head), too new (everything else but eigen), or `onnxruntime` is using a commit from an out-of-tree branch (eigen). My cmake foo is rather limited. If we could reuse the pre-build artifacts for the packages that are not vendored it would be great!

Reply:
Fair enough, I struggled with this as well when I tried packaging it ~2 year ago.